src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1407,9 +1536,17 @@ public WaitAnnotation(IResource resource, WaitType waitType, int exitCode = 0) {
 
         public IResource Resource { get { throw null; } }
 
+        public WaitBehavior? WaitBehavior { get { throw null; } init { } }
+
         public WaitType WaitType { get { throw null; } }
     }
 
+    public enum WaitBehavior
+    {
+        WaitOnDependencyFailure = 0,
+        StopOnDependencyFailure = 1
+    }
```

Comment:
@JamesNK You had thoughts on these names right?

@DamianEdwards @eerhardt @ReubenBond ?

Reply:
It isn't clear to me if `StopOnDependencyFailure` means:
- it will stop waiting if a dependency it's waiting on fails to start
- whether it throws an exception if the stop condition is hit
- it will stop the resource if the dependency fails after everything is started

Reply:
`FailOnDependencyFailure`

Reply:
I'd assume the behaviors we want to be able to select are:
- Keep waiting forever until the dependency has started
- Cancel waiting and throw an exception if the dependency fails to start

Do we think there's a need for being able to stop waiting and just start anyway if a dependency fails to start?

Reply:
Unclear if that last one is needed but it could be an option we add later.

Reply:
StopOnUnavailable

Reply:
I think continue is the behavior we don't have today.

Reply:
> `ContinueOnDependencyUnavailable`? Maybe `Stop` is too overloaded (with resource stopping, vs. stopping the wait operation)?

When the wait operation is stopped, isn't the waiting resource is stopped when `BeforeResourceStartedEvent` throws? Stop works in both ways.

Reply:
Yep it throws currently.

Reply:
That's not bad.

Reply:
@mitchdenny can this thread be resolved?

Reply:
The new names are:

```
public enum WaitBehavior
{
    /// <summary>
    /// If the resource is unavailable, continue waiting.
    /// </summary>
    WaitOnResourceUnavailable,

    /// <summary>
    /// If the resource is unavailable, stop waiting.
    /// </summary>
    StopOnResourceUnavailable
}
```

Which is different than @ReubenBond's suggestion above. Do we have an official consensus on what names we want here?

Reply:
I adopted the names from a branch that David had. In terms of enumeration name, I think `WaitBehavior` is fine. I do like `Wait` and `Throw` as enumeration values and am happy to make that change.

Reply:
I'm probably bikeshedding at this point (but hey, it's API review), but `WaitBehavior.Wait` and `WaitBehavior.Throw` kinda doesn't make sense. If it is `WaitBehavior` and the behavior is `Throw`, that isn't really "waiting".

Reply:
We closed on this.